### PR TITLE
Implemented solido state verification. 

### PR DESCRIPTION
**Pull request by Shi-eld**
Implemented solido state verification. Refactored verify_transaction script.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -181,10 +181,6 @@ jobs:
         cargo clippy --manifest-path program/Cargo.toml -- --deny warnings
         cargo clippy --manifest-path testlib/Cargo.toml -- --deny warnings
 
-    - name: Typecheck Python
-      run: |
-        git ls-files | grep '\.py$' | xargs mypy --strict
```
Comment (kkonevets) [inline .github/workflows/build.yml:186]: Why did you remove these lines? Please, remove only `--strict`

**Review by kkonevets — CHANGES_REQUESTED** [review]

File: scripts/migrate.sh [hunk 2]
```diff
@@ -37,7 +38,7 @@ solana withdraw-from-vote-account test-ledger/vote-account-keypair.json \
 ###############################################################################
 
 # create new v2 accounts
-../solido/target/debug/solido \
+../solido_v2/target/release/solido \
```
Comment (kkonevets) [inline scripts/migrate.sh:41]: You should keep a `debug` instead, because this script is for testing and quick checking purposes only. But you can edit user instructions the way you wish

File: scripts/migrate.sh [hunk 3]
```diff
@@ -18,13 +18,14 @@ solana withdraw-from-vote-account test-ledger/vote-account-keypair.json \
 # create instance
 ./tests/deploy_test_solido.py --verbose
 
+cp ../solido_test.json ../solido_config.json
```
Comment (kkonevets) [inline scripts/migrate.sh]: This script is for testing, so it should always be `./solido_test.json`. Please don't use this script for anything other than for testing.

File: scripts/migrate.sh [hunk 4]
```diff
@@ -18,13 +18,14 @@ solana withdraw-from-vote-account test-ledger/vote-account-keypair.json \
...
 # start maintainer
-./target/debug/solido --config ../solido_test.json \
-                      --keypair-path ../solido_old/tests/.keys/maintainer.json \
+./solido_v1/target/release/solido --config ./solido_config.json \
```
Comment (kkonevets) [inline scripts/migrate.sh:25]: Why did you rename the directory? Now it's not convenient to develop in debug mode. You should keep a `debug` instead, because this script is for testing. But you can edit user instructions the way you wish

File: scripts/migrate.sh [hunk 5]
```diff
@@ -110,28 +120,36 @@ awk '/{/,/}/' temp | jq -r .transaction_address >> output
...
     --vote-accounts validators.txt \
-    --keypair-path ../solido_old/tests/.keys/maintainer.json
+    --keypair-path ./solido_v1/tests/.keys/maintainer.json
+
+    ./script_update/scripts/operation.py \
+    check-transactions --phase adding --transactions-path adding_trx.txt
+
+# verify transaction
+./solido_v2/scripts/operation.py \
+    check-transactions --phase adding --transactions-path adding_trx.txt
```
Comment (kkonevets) [inline scripts/migrate.sh]: duplicated lines?

File: scripts/operation.py [hunk 6]
```diff
@@ -11,13 +11,21 @@
...
 sys.path.append(os.path.dirname(SCRIPT_DIR))
 
 from tests.util import solido, solana, run  # type: ignore
 
 
+def set_solido_cli_path(strData):
+    if os.path.isfile(strData):
+        os.environ["SOLPATH"] = strData
+    else:
+        print("Program not exist: " + strData)
```
Comment (kkonevets) [inline scripts/operation.py]: `Program does not exist`

File: scripts/operation.py [hunk 7]
```diff
@@ -104,38 +99,50 @@ def get_signer() -> Any:
...
+    install_solido()
+    with open(os.getenv("SOLIDO_CONFIG")) as f:
         config = json.load(f)
         cluster = config.get("cluster")
         if cluster:
             os.environ['NETWORK'] = cluster
 
     if args.command == "deactivate-validators":
-        lido_state = solido('--config', args.config, 'show-solido')
+        set_solido_cli_path(os.getenv("SOLIDO_V1"))
```
Comment (kkonevets) [inline scripts/operation.py:137]: This is not a good idea. We should not use hard coded environment variables. They should be passes as CLI arguments.
And I could run this script from any directory, by default it should be a current directory. And I don't have to specify env variable. Instead I could specify a CLI argument `--base_path path/to/solido_v1`. It is much clear than env variables

File: scripts/operation.py [hunk 8]
```diff
@@ -104,38 +99,50 @@ def get_signer() -> Any:
...
+
+    current_parser = subparsers.add_parser('test', help='`Command for tests`')
 
     args = parser.parse_args()
 
     sys.argv.append('--verbose')
 
-    with open(args.config) as f:
+    install_solido()
+    with open(os.getenv("SOLIDO_CONFIG")) as f:
```
Comment (kkonevets) [inline scripts/operation.py:129]: This is not a good idea. We should not use hard coded environment variables. You could just pass it as an argument 
```python
lido_state = solido('--config', os.getenv("SOLIDO_CONFIG"), 'show-solido')
```

File: scripts/operation.py [hunk 9]
```diff
@@ -45,10 +50,7 @@ def get_signer() -> Any:
         required=True,
     )
     current_parser.add_argument(
-        "--outfile",
-        type=str,
-        help='Output file path',
-        required=True,
+        "--outfile", type=str, help='Output file path', required=True
```
Comment (kkonevets) [inline scripts/operation.py:54]: You should use same (as previously used) code formatting to avoid reformatting and cluttering.

File: scripts/migrate.sh (hunk 4 above)
Comment (Shi-eld) [reply scripts/migrate.sh:25]: 1. The previous folder names were poorly chosen and confused the users of the scripts. The correct framework is to assign unique names.
2. It's not good practice to use debug builds in user scripts. All user scripts must target on the release build. We can create another script with a _debug postfix to indicate to users that this script is a helper for developers.

File: scripts/operation.py (hunk 7 above)
Comment (Shi-eld) [reply scripts/operation.py:137]: The option `--config` has been hidden to reduce the load on multisig owners and simplify instructions for them.
A good interface design framework here is simplification to reduce the number of errors in use.
